pkg/volume/volume.go

```diff
@@ -0,0 +1,316 @@
+	if err != nil {
+		klog.Errorf("failed ResizeVolume %v", err)
+		return response, err
+	}
+	return response, nil
+}
+
+func (v *Volume) VolumeStats(context context.Context, request *VolumeStatsRequest) (*VolumeStatsResponse, error) {
+	getVolumeStatsRequest := &GetVolumeStatsRequest{
+		VolumeId: request.VolumeId,
+	}
+	getVolumeStatsResponse, err := v.GetVolumeStats(context, getVolumeStatsRequest)
+	if err != nil {
+		return nil, fmt.Errorf("Forward to GetVolumeStats failed, err=%+v", err)
+	}
+	volumeStatsResponse := &VolumeStatsResponse{
+		VolumeSize:     getVolumeStatsResponse.TotalBytes,
+		VolumeUsedSize: getVolumeStatsResponse.UsedBytes,
+	}
+	return volumeStatsResponse, nil
+}
+
+func (v *Volume) GetVolumeStats(context context.Context, request *GetVolumeStatsRequest) (*GetVolumeStatsResponse, error) {
+	klog.V(2).Infof("GetVolumeStats: Request: %+v", request)
+	volumeID := request.VolumeId
+	if volumeID == "" {
+		return nil, fmt.Errorf("volume id empty")
+	}
+
+	totalBytes, usedBytes, err := v.hostAPI.GetVolumeStats(volumeID)
+	if err != nil {
+		klog.Errorf("failed GetVolumeStats %v", err)
+		return nil, err
+	}
+
+	klog.V(2).Infof("VolumeStats: returned: Capacity %v Used %v", totalBytes, usedBytes)
+
+	response := &GetVolumeStatsResponse{
+		TotalBytes: totalBytes,
+		UsedBytes:  usedBytes,
+	}
+
+	return response, nil
+}
+
+func (v *Volume) GetVolumeDiskNumber(context context.Context, request *VolumeDiskNumberRequest) (*VolumeDiskNumberResponse, error) {
+	getDiskNumberFromVolumeIDRequest := &GetDiskNumberFromVolumeIDRequest{
+		VolumeId: request.VolumeId,
+	}
+	getDiskNumberFromVolumeIDResponse, err := v.GetDiskNumberFromVolumeID(context, getDiskNumberFromVolumeIDRequest)
+	if err != nil {
+		return nil, fmt.Errorf("Forward to GetDiskNumberFromVolumeID failed, err=%+v", err)
+	}
+	volumeStatsResponse := &VolumeDiskNumberResponse{
+		DiskNumber: int64(getDiskNumberFromVolumeIDResponse.DiskNumber),
+	}
+	return volumeStatsResponse, nil
+}
+
+func (v *Volume) GetDiskNumberFromVolumeID(context context.Context, request *GetDiskNumberFromVolumeIDRequest) (*GetDiskNumberFromVolumeIDResponse, error) {
+	klog.V(2).Infof("GetDiskNumberFromVolumeID: Request: %+v", request)
+
+	volumeId := request.VolumeId
+	if volumeId == "" {
+		return nil, fmt.Errorf("volume id empty")
+	}
+
+	diskNumber, err := v.hostAPI.GetDiskNumberFromVolumeID(volumeId)
+	if err != nil {
+		klog.Errorf("failed GetDiskNumberFromVolumeID %v", err)
+		return nil, err
+	}
+
+	response := &GetDiskNumberFromVolumeIDResponse{
+		DiskNumber: diskNumber,
+	}
+
+	return response, nil
+}
+
+func (v *Volume) GetVolumeIDFromMount(context context.Context, request *VolumeIDFromMountRequest) (*VolumeIDFromMountResponse, error) {
+	getVolumeIDFromTargetPathRequest := &GetVolumeIDFromTargetPathRequest{
+		TargetPath: request.Mount,
+	}
+	getVolumeIDFromTargetPathResponse, err := v.GetVolumeIDFromTargetPath(context, getVolumeIDFromTargetPathRequest)
```

Comment:
Looks like this is compat wrapper as well. Consider dropping `GetVolumeIDFromMount`.

Reply:
Yeah! I just talked to @mauriciopoppe and discovered what these extra methods are. The plan is to have a PR to drop all the compat methods after the initial API group migrations (see #241). I think it's easier to just copy paste the 
current implementation in the first pass.